website/addons/googledrive/model.py

```diff
@@ -111,218 +101,183 @@ def get_or_create(cls, node, path):
         return new, created
 
 
-class GoogleDriveOAuthSettings(StoredObject):
-    """
-    this model address the problem if we have two osf user link
-    to the same google drive user and their access token conflicts issue
-    """
+class GoogleDriveProvider(ExternalProvider):
+    name = 'Google Drive'
+    short_name = 'googledrive'
+
+    client_id = drive_settings.CLIENT_ID
+    client_secret = drive_settings.CLIENT_SECRET
+
+    auth_url_base = "https://accounts.google.com/o/oauth2/auth?access_type=offline&approval_prompt=force"
+    callback_url = 'https://www.googleapis.com/oauth2/v3/token'
+
+    default_scopes = drive_settings.OAUTH_SCOPE
+    _auth_client = GoogleAuthClient()
+    _drive_client = GoogleDriveClient()
+
+    def handle_callback(self, response):
+        client = self._auth_client
+        info = client.userinfo(response['access_token'])
+        return {
+            'provider_id': info['sub'],
+            'display_name': info['name'],
+            'profile_url': info.get('profile', None)
+        }
 
-    # google drive user id, for example, "4974056"
-    user_id = fields.StringField(primary=True, required=True)
-    # google drive user name this is the user's login
-    username = fields.StringField()
-    access_token = fields.StringField()
-    refresh_token = fields.StringField()
-    expires_at = fields.DateTimeField()
+    # TODO: Remove, if not used
+    def _get_folders(self):
+        """ Get a list of a user's folders"""
+        client = self._drive_client
+        return client.folders()
+
+    def _refresh_token(self, access_token, refresh_token):
+        client = self._auth_client
+        if refresh_token:
+            token = client.refresh(access_token, refresh_token)
+            return token
+        else:
+            exceptions.AddonError("Refresh Token is not Obtained")
```

Comment:
This does not actually raise the exception.

Additionally, it would help if there were docstrings explaining the difference in purpose between `_refresh_token` and `refresh_access_token`; the names are confusingly similar.